### PR TITLE
Fix provision; convert doc

### PR DESCRIPTION
**Pull request by stevemessick**
@devoncarew @jacob314 @pq 

Fix various bugs in provisioning. Make it more resilient to errors.

It can still get into a bad state. The simplest fix is:
```shell
rm -rf artifacts
plugin build
```
Also, I converted the documentation to markdown and added it. I couldn't find an easy way to wrap the lines so the raw markdown is unreadable. However, it looks good in a proper reader. As I update it I will wrap the lines.

### CODE REVIEW

File: tool/plugin/lib/plugin.dart [hunk 1]
```diff
@@ -432,6 +432,13 @@ class ArtifactManager {
           log('download failed');
           break;
         }
+        var archiveFile = new File(path);
+        if (!archiveFile.existsSync() || archiveFile.lengthSync() < 200) {
+          log('archive file not found: $base/${artifact.file}');
```
Comment (jacob314) [inline tool/plugin/lib/plugin.dart]: nit: why do files less than length 200 need to be treated as missing?

Comment (stevemessick) [reply tool/plugin/lib/plugin.dart]: If the file is missing then the downloader returns an XML file, length 127, that contains an error message. I considered analyzing the XML to make sure it was reporting a missing-file error, but didn't bother. The files are all quite large.

**Comment by jacob314** [comment]
Makes sense. Might want to add a comment explaining why 200 is used for the
next person who touches the code.

**Comment by stevemessick** [comment]
Good point! Done locally.

**Pull request merged**